store/tikv/latch/latch.go

```diff
@@ -33,120 +33,161 @@ type Latch struct {
 	sync.Mutex
 }
 
-func (l *Latch) occupied() bool {
+func (l *latch) occupied() bool {
 	return l.waitingQueueHead != 0
 }
 
-func (l *Latch) free() {
+func (l *latch) free() {
 	l.waitingQueueHead = 0
 }
 
+func (l *latch) refreshCommitTS(commitTS uint64) {
+	l.Lock()
+	defer l.Unlock()
+	l.maxCommitTS = mathutil.MaxUint64(commitTS, l.maxCommitTS)
+}
+
 // Lock is the locks' information required for a transaction.
 type Lock struct {
 	// The slot IDs of the latches(keys) that a startTS must acquire before being able to processed.
 	requiredSlots []int
 	// The number of latches that the transaction has acquired.
 	acquiredCount int
-	// Whether current transaction is waiting
-	isWaiting bool
 	// Current transaction's startTS.
 	startTS uint64
+	// Current transaction's commitTS.
+	commitTS uint64
+
+	wg     sync.WaitGroup
+	status acquireResult
```

Comment:
This field will also be used to check the Lock's status after `Lock`. I think is thread-safe, since, for a transaction, the following function would never be called at the same time
1. `acquire` && `release`
2. two or more `acquire` or `release`